app/controllers/settings/profiles_controller.rb

```diff
@@ -14,17 +14,19 @@ class Settings::ProfilesController < ApplicationController
   def show; end
 
   def update
-    if @account.update(account_params)
-      redirect_to settings_profile_path, notice: I18n.t('generic.changes_saved_msg')
-    else
+    account_update = account_params
+
+    if account_update.nil? || !@account.update(account_update)
       render :show
+    else
+      redirect_to settings_profile_path, notice: I18n.t('generic.changes_saved_msg')
     end
   end
 
   private
 
   def account_params
-    params.require(:account).permit(:display_name, :note, :avatar, :header, :locked)
+    params[:account]&.permit(:display_name, :note, :avatar, :header, :locked)
```

Comment:
What are you trying to do here?

Is the concern that `params[:account]` can be nil, and you want to handle that case and the case where it has normal values via strong params?

How about something like: `params.fetch(:account, {}).permit(:display_name, ...)`